code/processes/gateway.q

```diff
@@ -409,6 +413,12 @@ getserverids:{[att]
   ];
 
   // its a dictionary of attributes
+  // check if attribute types are correct types
+  correctAttTypes:`dates`servertypes`tables!(14h;11h;11h);
+  w:(correctAttTypes [key att]=type each att);
+  if[not all{[x;y] y each key x}[att;w];
+  '("Wrong function parameter types provided.", " ",(raze " " sv string (where not w))," parameter(s) need type(s) ",raze " " sv string correctAttTypes where not w),"h"
```

Comment:
Fix indent (+2 spaces). Should the first `where not w` be used in indexing? Seems like a stringed Boolean list will be returned here.

Reply:
`where not w` should return
`,`dates`